### PR TITLE
WIZZIE-2305 Develop normalizer group regex function

### PR DESCRIPTION
**Pull request by andresgomezfrr**

### CODE REVIEW

File: docs/_posts/2017-02-14-mapper-functions.md [hunk 1]
```diff
@@ -992,3 +993,45 @@ If we use this message using the ArrayDecompositionMapper that is defined on the
   "dim3":"C"
 }
 ```
+
+### <a name="regexMapper"></a> RegexMapper [ [Top](#index) ]
+
+The RegexMapper is a function that allows us to extrar new dimension from Strig using regex rules.
```
Comment (pablocasares) [inline docs/_posts/2017-02-14-mapper-functions.md]: extract a new dimension from String

File: docs/_posts/2017-02-14-mapper-functions.md [hunk 2]
```diff
@@ -992,3 +993,45 @@ If we use this message using the ArrayDecompositionMapper that is defined on the
...
+
+```json
+        {
+          "name": "myRegexMapper",
+          "className": "io.wizzie.normalizer.funcs.impl.RegexMapper",
+          "properties": {
+            "generateDimensions":["dim1", "dim2", "dim3"],
```
Comment (pablocasares) [inline docs/_posts/2017-02-14-mapper-functions.md]: Is this property necessary? I think you can use the group names. What do you think?

File: docs/_posts/2017-02-14-mapper-functions.md [hunk 3]
```diff
@@ -992,3 +993,45 @@ If we use this message using the ArrayDecompositionMapper that is defined on the
...
+            "regexPattern": "test test (?<dim1>.*) test (?<dim2>.*) test (?<dim3>.*) test",
+            "parserDimension": "message"          }
```
Comment (pablocasares) [inline docs/_posts/2017-02-14-mapper-functions.md]: I think "parseDimension" is more explicit. what do you think?

File: docs/_posts/2017-02-14-mapper-functions.md [hunk 4]
```diff
@@ -992,3 +993,45 @@ If we use this message using the ArrayDecompositionMapper that is defined on the
...
+        }
+```  
+
+The RegexMapper has three properties:
+
+ * `generateDimensions`: The new dimensions generate from the String
```
Comment (pablocasares) [inline docs/_posts/2017-02-14-mapper-functions.md]: generated

File: docs/_posts/2017-02-14-mapper-functions.md [hunk 5]
```diff
@@ -992,3 +993,45 @@ If we use this message using the ArrayDecompositionMapper that is defined on the
...
+ * `regexPattern`: The regex pattern to extrac the dimensions. You must use groups `(...)` and group names. `?<...>`
```
Comment (pablocasares) [inline docs/_posts/2017-02-14-mapper-functions.md]: to extract

File: docs/_posts/2017-02-14-mapper-functions.md (hunk 2 above)
Comment (andresgomezfrr) [reply docs/_posts/2017-02-14-mapper-functions.md]: Java doesn't support get the available dimensions from the regex. You can get the group value using the group name inside `?<...>`, but you can't discover all the available regex group name

Comment (pablocasares) [reply docs/_posts/2017-02-14-mapper-functions.md]: ok, perfect.

**Pull request merged**